packages/react-scripts/config/modules.js

```diff
@@ -0,0 +1,145 @@
+// @remove-on-eject-begin
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+// @remove-on-eject-end
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const chalk = require('chalk');
+const paths = require('./paths');
+
+/**
+ * Get the baseUrl of a compilerOptions object.
+ *
+ * @param {Object} options
+ */
+function getAdditionalModulePath(options = {}) {
+  const baseUrl = options.baseUrl;
+
+  // We need to explicitly check for null and undefined (and not a falsy value) because
+  // TypeScript treats an empty string as `.`.
+  if (baseUrl == null) {
+    return null;
+  }
+
+  const baseUrlResolved = path.resolve(paths.appPath, baseUrl);
+
+  // We don't need to do anything if `baseUrl` is set to `node_modules`. This is
+  // the default behavior.
+  if (path.relative(paths.appNodeModules, baseUrlResolved) === '') {
+    return null;
+  }
+
+  // Allow the user set the `baseUrl` to `appSrc`.
+  if (path.relative(paths.appSrc, baseUrlResolved) === '') {
+    return paths.appSrc;
+  }
+
+  // Otherwise, throw an error.
+  throw new Error(
+    chalk.red.bold(
+      "Your project's `baseUrl` can only be set to `src` or `node_modules`." +
+        ' Create React App does not support other values at this time.'
+    )
+  );
+}
+
+/**
+ * Get the alias of a compilerOptions object.
+ *
+ * @param {Object} options
+ */
+function getAliases(options = {}) {
```

Comment:
This function still seems a bit restrictive -- can we allow multiple entries?

Reply:
It does support multiple entries, e.g.
```json
{
  "compilerOptions": {
    "baseUrl": "src",
    "paths": {
      "@": ["src"],
      "~": ["src"]
    }
  }
}
```

Reply:
Ah, we should let them map it to more than just `src`.

Reply:
Ah misunderstood you then. But only `src` or subfolders of `src` then? As in `path.relative(paths.appSrc, aliasPath) === '' || path.relative(paths.appSrc, aliasPath).startsWith('./')`

Reply:
Yeah, subfolders of `src` or `src` itself.